eval.c

```diff
@@ -259,9 +260,10 @@ static sexp sexp_flatten_dot (sexp ctx, sexp ls) {
 
 int sexp_param_index (sexp ctx, sexp lambda, sexp name) {
   sexp ls;
-  int i;
+  int i = 0;
   while (1) {
-    i = 0;
+    // ISSUE_296 - Value stored to 'i' is never used
+    //i = 0;
```

Comment:
That comment is confusing... 'i' is indeed used, it's the return value.
Moreover moving the initialization outside the loop breaks the logic.